server/command_list.go

```diff
@@ -28,23 +28,46 @@ func (p *Plugin) runListCommand(args []string, extra *model.CommandArgs) (*model
 }
 
 func (p *Plugin) getUpdatedInstallsForUser(userID string) ([]*Installation, error) {
-	installsForUser, err := p.getInstallationsForUser(userID)
+	pluginInstalls, err := p.getInstallationsForUser(userID)
 	if err != nil {
 		return nil, err
 	}
 
-	var updatedInstall *cloud.Installation
-	for _, install := range installsForUser {
-		updatedInstall, err = p.cloudClient.GetInstallation(install.ID)
+	var cloudInstall *cloud.Installation
+	for k, pluginInstall := range pluginInstalls {
+		cloudInstall, err = p.cloudClient.GetInstallation(pluginInstall.ID)
 		if err != nil {
-			return nil, errors.Wrapf(err, "could not get updated installation %s", install.ID)
+			return nil, errors.Wrapf(err, "could not get updated installation %s", pluginInstall.ID)
 		}
-		if updatedInstall == nil {
-			return nil, fmt.Errorf("could not find installation %s", install.ID)
+
+		if cloudInstall == nil {
+			// If it was never an installation in cloud server and it was never marked as deleted
+			// in KV store, we return en error.
+			if pluginInstall.DeleteAt == 0 {
```

Comment:
Hmm, I am not sure that this is correct. The deleteAt timestamp here is a value remembered by the plugin database. I am not sure why we would want to check this older value.